src/main/java/com/rcv/CommonDataFormatReader.java

```diff
@@ -0,0 +1,162 @@
+      // we expect a top-level "CVR" object containing a list of CVR objects
+      ArrayList CVRs = (ArrayList) json.get("CVR");
+      // for each CVR object extract the current snapshot
+      // it will be used to build the ballot data
+      for (Object CVR : CVRs) {
+        HashMap CVRObject = (HashMap) CVR;
+        String currentSnapshotID = (String) CVRObject.get("CurrentSnapshotId");
+        // get ballotID
+        String ballotID = (String) CVRObject.get("BallotPrePrintedId");
+        // compute internal ballot ID
+        String computedCastVoteRecordID = String.format("%s(%d)", fileName, ++cvrIndex);
+
+        ArrayList CVRSnapshots = (ArrayList) CVRObject.get("CVRSnapshot");
+        for (Object snapshotObject : CVRSnapshots) {
+          HashMap snapshot = (HashMap) snapshotObject;
+          if (snapshot.get("@id").equals(currentSnapshotID)) {
+            // we found the current CVR snapshot
+            // parse out the rankings and create a new cvr
+
+            // list to contain rankings as they are parsed
+            List<Pair<Integer, String>> rankings = new ArrayList<>();
+
+            // at the top level is a list of contests each of which contains selections
+            ArrayList CVRContests = (ArrayList) snapshot.get("CVRContest");
+            for (Object contestObject : CVRContests) {
+              // extract the CVRContest
+              HashMap CVRContest = (HashMap) contestObject;
+              // contest contains contestSelections
+              ArrayList contestSelections = (ArrayList) CVRContest
+                  .get("CVRContestSelection");
+              for (Object contestSelectionObject : contestSelections) {
+                // extract the contestSelection
+                HashMap contestSelection = (HashMap) contestSelectionObject;
+                // selectionID is the candidate/contest ID for this selection position
+                String selectionID = (String) contestSelection.get("ContestSelectionId");
+                // extract all the positions (ranks) which this selection has been assigned
+                ArrayList selectionPositions = (ArrayList) contestSelection
+                    .get("SelectionPosition");
+                for (Object selectionPositionObject : selectionPositions) {
+                  // extract the position object
+                  HashMap selectionPosition = (HashMap) selectionPositionObject;
+                  // and finally the rank
+                  Integer rank = (Integer) selectionPosition.get("Rank");
+                  assert rank != null;
+                  // create a new ranking object and save it
+                  rankings.add(new Pair<>(rank, selectionID));
+                }
+              }
+            }
+
+            // create new cast vote record
+            CastVoteRecord newRecord = new CastVoteRecord(computedCastVoteRecordID,
+                ballotID, null, null, rankings);
```

Comment:
I guess there's no `fullCVRData` equivalent here? Will our existing audit log just not be compatible with CDF input?

Reply:
The idea is that CDF _is_ the audit log.